pkg/storage/periodical_archiver.go

```diff
@@ -0,0 +1,129 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"sync"
+	"time"
+)
+
+type aggregationFunction func(items []interface{}) (interface{}, error)
```

Comment:
help would be really useful. Here for `aggregationFunction` and/or bellow - this whole package handles abstract (`interface{}`) types which makes harder for me to understand it.